apps/api/migrations/064_create_vault_rebalance_legs.up.sql

```
@@ -0,0 +1,18 @@
+    id                     UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    rebalance_id           UUID REFERENCES vault_rebalances(id) ON DELETE CASCADE,
+    vault_contract_address TEXT NOT NULL CHECK (char_length(vault_contract_address) > 0),
+    plan_hash              TEXT,
+    source_id              TEXT NOT NULL CHECK (char_length(source_id) > 0),
+    delta                  NUMERIC(38,0) NOT NULL,
+    amount_out             NUMERIC(38,0) NOT NULL DEFAULT 0,
+    min_out                NUMERIC(38,0) NOT NULL DEFAULT 0,
+    occurred_at            TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+CREATE INDEX IF NOT EXISTS idx_vault_rebalance_legs_vault ON vault_rebalance_legs (vault_contract_address, occurred_at DESC);
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)